python/tk_multi_workfiles/errors.py

```diff
@@ -0,0 +1,63 @@
+# Copyright (c) 2016 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+"""
+Workfiles 2 related errors.
+"""
+
+from sgtk import TankError
+
+
+class WorkfilesError(TankError):
+    """
+    Base class for work area related errors.
+    """
+
+
+class MissingTemplatesError(WorkfilesError):
+    """
+    Raised when one or more templates are missing.
+    """
+
+    def __init__(self, missing_templates):
+        """
+        Constructor.
+
+        :param missing_templates: List of templates that are missing.
+        """
+        WorkfilesError.__init__(self, self.generate_missing_templates_message(missing_templates))
+
+    @classmethod
+    def generate_missing_templates_message(self, missing_templates):
+        """
+        Generates a warning for when templates are not all configured.
+        """
+        if len(missing_templates) == 4:
```

Comment:
We shouldn't have 4 here as a magic number... put a constant somewhere for the # of templates or something else that'll be easier to catch if we ever add (or remove) a template from the settings?